### PR TITLE
Extend Invoice resource

### PR DESCRIPTION
**Pull request by iranl**
- Adds listAll() function to Invoices
- Adds results/messages to Invoices through BaseObject
- Adds VAT lines
- Adds UnitsPriceInc to Sales invoice lines (for Invoice types that include VAT)
- Adds customerName, debitCredit and invoiceAmount variables which are returned when using listAll()
- Changes looping through Sales invoice lines from:
```php
$responseDOM->getElementsByTagName('line')
```
To:

```php
$linesDOM->childNodes
``` 

- Some formatting/order changes

### CODE REVIEW

File: src/Invoice.php [hunk 1]
```diff
@@ -80,6 +101,28 @@ public function setCustomer(Customer $customer)
...
+        return $this->customerName;
+    }
+
+    public function setCustomerName($customerName)
+    {
+        $this->customerName = $customerName;
+        return $this;
+    }
+
+    public function getDebitCredit()
```
Comment (willemstuursma) [inline src/Invoice.php]: Is this some kind of boolean or enum? Can you add types please?

File: src/Invoice.php [hunk 2]
```diff
@@ -91,6 +134,17 @@ public function getTotals(): InvoiceTotals
         return $this->totals;
     }
 
+    public function getInvoiceAmount()
```
Comment (willemstuursma) [inline src/Invoice.php]: Can you make this work with `Money\Money`?

File: src/Invoice.php [hunk 3]
```diff
@@ -146,6 +200,17 @@ public function setInvoiceDate($invoiceDate)
         return $this;
     }
 
+    public function getDueDate()
```
Comment (willemstuursma) [inline src/Invoice.php]: Can you make this work with DateTimeInterface types?

File: src/InvoiceVatLine.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    private $vatValue;
+    private $performanceDate;
+
+    /**
+     * @var PerformanceType|null Mandatory in case of an ICT VAT code.
+     */
+    private $performanceType;
+
+    public function getVatValue()
```
Comment (willemstuursma) [inline src/InvoiceVatLine.php]: Is this a monetary amount?

File: src/InvoiceVatLine.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        return $this->vatValue;
+    }
+
+    public function setVatValue($vatValue)
+    {
+        $this->vatValue = $vatValue;
+        return $this;
+    }
+
+    public function getPerformanceDate()
```
Comment (willemstuursma) [inline src/InvoiceVatLine.php]: Can you change this to DateTimeInterface?

File: src/InvoiceVatLine.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        return $this->performanceDate;
+    }
+
+    public function setPerformanceDate($performanceDate)
```
Comment (willemstuursma) [inline src/InvoiceVatLine.php]: Should the PerformanceType be moved to a dedicated trait?

**Comment by iranl** [comment]
> Thanks. I would like to use a bit more typing where appropriate. Would it be possible to add some new test cases?

Thanks for the review and requested changes, will get on them in the coming week.

File: src/Invoice.php (hunk 1 above)
Comment (iranl) [reply src/Invoice.php]: Fixed in #142

File: src/Invoice.php (hunk 3 above)
Comment (iranl) [reply src/Invoice.php]: Fixed in #142

File: src/Invoice.php (hunk 2 above)
Comment (iranl) [reply src/Invoice.php]: Fixed in #142

File: src/InvoiceVatLine.php (hunk 4 above)
Comment (iranl) [reply src/InvoiceVatLine.php]: Fixed in #142

File: src/InvoiceVatLine.php (hunk 5 above)
Comment (iranl) [reply src/InvoiceVatLine.php]: Fixed in #142

File: src/InvoiceVatLine.php (hunk 6 above)
Comment (iranl) [reply src/InvoiceVatLine.php]: Fixed in #142

**Comment by iranl** [comment]
Superseded by #142

**Pull request closed without merging**